pueblo/ngr/runner.py

```diff
@@ -455,22 +453,66 @@ def test(self) -> None:
         """
         Test a Python thing, based on which test runner is installed.
 
-        TODO: Figure out how to invoke target `poe check`, which is popular amongst
-              users of `pueblo`. It bundles linter/checkstyle and software test targets
-              into a single entrypoint, similar to what `gradle check` is doing.
+        First, try a high-level target of some sort of management tool,
+        discover its targets, and invoke `check` or `test`.
+
+        When no high-level tool can be discovered, just invoke `pytest`.
         """
 
+        # On Poetry projects, invoke poetry/pytest.
+        # TODO: Also allow using Poetry with other task runners, see below.
         if self.has_poetry_lock:
             # TODO: poetry run which pytest
             run_command("poetry run pytest --config-file=.")
 
         else:
+            uses_poe = self.has_pyproject_toml and self.pyproject_has_string("[tool.poe.tasks]")
```

Comment:
```suggestion
            uses_poe = self.has_pyproject_toml and self.pyproject_has_string("[tool.poe.tasks")
```
the tasks might be defined like `[tool.poe.tasks.test]`

Reply:
Ah. Thanks for noticing. I already expected this to be a bit too sloppy. I think we will also use poethepoet right away instead of bothering to parse manually.

Reply:
I see, well you could probably use the PoeConfig class for most of it, though beware I don't consider it an external API at this point, in fact I'm currently in the middle of [a refactor](https://github.com/nat-n/poethepoet/tree/refactor_tasks) that _might_ impact that API a bit.